packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue, useIdAllocator } from '@leafygreen-ui/hooks';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import { Label } from '@leafygreen-ui/typography';
+
+import { InputIcon } from '../InputIcon';
+import { StateNotifications } from '../StateNotifications';
+import { TogglePassword } from '../TogglePassword';
+
+import {
+  inputBaseStyles,
+  inputBaseThemeStyles,
+  inputDisabledBaseStyles,
+  inputDisabledThemeStyles,
+  inputIconSizeStyles,
+  inputSizeStyles,
+  inputThemeStyles,
+  inputWrapperStyles,
+  labelBaseStyles,
+  labelLargeOverrideStyles,
+} from './PasswordInput.styles';
+import {
+  type NotificationProps,
+  PasswordInputProps,
+  SizeVariant,
+  States,
+} from './PasswordInput.types';
+import { getStateFromArray } from './utils';
+
+export const PasswordInput = React.forwardRef<
+  HTMLInputElement,
+  PasswordInputProps
+>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      darkMode: darkModeProp,
+      id: idProp,
+      'aria-describedby': ariaDescribedbyProp,
+      'aria-labelledby': ariaLabelledbyProp,
+      sizeVariant = SizeVariant.Default,
+      stateNotifications = [],
+      disabled = false,
+      autoComplete = 'new-password',
+      className,
+      label,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'lg-passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const descriptionId = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const labelId = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
```

Comment:
ariaLabelledby means that a label exists in the DOM so I actually don't think this is what we want to be setting the default id to

Reply:
oh good catch, i mixed up the names when i was addressing the first round of feedback

Reply:
oh wait, i think this is correct, if they are passing ariaLabelledby then we want to pass that id to the input attribute `aria-labelledby` and idAllocator will not generate a new id. If ariaLabelledby is not passed then we generate an id and pass that to the input attribute `aria-labelledby` and use that as the `<Label>` id.

Reply:
As far as the label id goes, this looks correct (as long as the id allocator respects when `ariaLabelledbyProp` is `undefined`)

Got me thinking though, is there any difference between using `id` + `labelledby` vs `htmlFor` + `id`?
i.e. 
```html
<label id="foo"> <input aria-labelledby="foo">
vs
<label for="foo"> <input id="foo">
```

Reply:
Actually we shouldn't need this, since the existence of `ariaLabelledbyProp` implies that there is an external element that is labelling the input. It would be an anti-pattern to pass in both `aria-labelled` by and a value for `label`. 
Regardless of what we decide, we should add documentation for this in the README

Reply:
To your first point, I did some [quick research](https://www.w3.org/WAI/tutorials/forms/labels/) and it looks like I can just get away with `htmlFor` and `id` -- I don't need `labelId`. 

This is how it would work if we got rid of `labelId` and `ariaLabelledby`:
- If a label is passed in then the label will show. The input `id` will be used for the labels `for` attribute; that's good enough.
- if no label is passed then its up to the consumer to pass in `ariaLabelledby`. That will get added as an attribute to the `input`.
- if both `label` and `ariaLabelledby` are passed in, I can update the code so that the `label` will not show up but the `ariaLabelledby` attribute will be added to the input.

Reply:
I think what you describe here makes sense, though I'd think showing `label` would be a better way to go for point 3.

I don't see this implemented though.

As it stands, if a consumer passes in both a `label` and `ariaLabelledby` prop, there will be two elements in the DOM with the same `id`: our label element and whatever external element the consumer adds the id to

Reply:
oh i missed this, i'll update